src/retcodes.jl

```diff
@@ -348,6 +376,10 @@ function Base.convert(::Type{ReturnCode.T}, retcode::Symbol)
         ReturnCode.ConvergenceFailure
     elseif retcode == :Failure || retcode == symfalse
         ReturnCode.Failure
+    elseif retcode == :Infeasible || retcode == :INFEASIBLE ||
+           retcode == :DUAL_INFEASIBLE || retcode == :LOCALLY_INFEASIBLE ||
+           retcode == :INFEASIBLE_OR_UNBOUNDED
```

Comment:
note that this conversion will become no longer automatic rather soon. I think we can keep it around as `symbol_to_retcode(retcode)`, but it won't be `convert` soon (since that invalidates everything).

Reply:
OK, but that is just a change to the solution constructor later?

Reply:
I guess we could put it in the solution constructor? I hadn't actually thought about that. For everything except optimization, the conversion from Symbols is being entirely eliminated. It would probably be best to do this conversion in the MOI wrapper script, but we can do a slow move to that so it's never breaking.

Reply:
So a `_status_to_sciml_retcode` method in `Optimization.jl` and the solution only accepts `SciMLBase.ReturnCode`?

Reply:
that accepts the MOI enums and spits out SciMLBase.ReturnCodes